src/particles/pusher/PlasmaParticleAdvance.cpp

```diff
@@ -257,22 +263,26 @@ ResetPlasmaParticles (PlasmaParticleContainer& plasma, int const lev, const bool
 
         if (initial) plasma.UpdateDensityFunction();
         auto density_func = plasma.m_density_func;
-        const amrex::Real c_t = get_phys_const().c * Hipace::m_physical_time;
+        const amrex::Real c_t = phys_const.c * Hipace::m_physical_time;
 
-        amrex::ParallelFor(
+        amrex::ParallelForRNG(
             pti.numParticles(),
-            [=] AMREX_GPU_DEVICE (long ip) {
+            [=] AMREX_GPU_DEVICE (long ip, const amrex::RandomEngine& engine) {
 
                 amrex::ParticleReal xp, yp, zp;
                 int pid;
                 GetPosition(ip, xp, yp, zp, pid);
                 if (initial == false){
                     SetPosition(ip, x_prev[ip], y_prev[ip], zp);
                 } else {
+
+                    amrex::Real u[3] = {0.,0.,0.};
+                    ParticleUtil::get_gaussian_random_momentum(u, u_mean, u_std, engine);
+
                     SetPosition(ip, x0[ip], y0[ip], zp, std::abs(pid));
                     w[ip] = w0[ip] * density_func(x0[ip], y0[ip], c_t);
-                    uxp[ip] = 0._rt;
-                    uyp[ip] = 0._rt;
+                    uxp[ip] = u[0]*phys_const.c;
+                    uyp[ip] = u[1]*phys_const.c;
                     psip[ip] = 0._rt;
```

Comment:
Why no temperature contribution here?
```suggestion
                    psip[ip] = 0._rt;
```